backend/src/auth/ROLES.ts

```diff
@@ -1,40 +1,31 @@
+import { ADMIN_RIGHTS } from './ADMIN_RIGHTS'
 import { INALIENABLE_RIGHTS } from './INALIENABLE_RIGHTS'
-import { RIGHTS } from './RIGHTS'
+import { MODERATOR_RIGHTS } from './MODERATOR_RIGHTS'
 import { Role } from './Role'
+import { USER_RIGHTS } from './USER_RIGHTS'
 
-export const ROLE_UNAUTHORIZED = new Role('unauthorized', INALIENABLE_RIGHTS)
-export const ROLE_USER = new Role('user', [
+export enum ROLE_NAMES {
+  ROLE_NAME_UNAUTHORIZED = 'unauthorized',
+  ROLE_NAME_USER = 'user',
+  ROLE_NAME_MODERATOR = 'moderator',
+  ROLE_NAME_ADMIN = 'admin',
+}
+export const ROLE_UNAUTHORIZED = new Role(ROLE_NAMES.ROLE_NAME_UNAUTHORIZED, INALIENABLE_RIGHTS)
+export const ROLE_USER = new Role(ROLE_NAMES.ROLE_NAME_USER, [
   ...INALIENABLE_RIGHTS,
-  RIGHTS.VERIFY_LOGIN,
-  RIGHTS.BALANCE,
-  RIGHTS.LIST_GDT_ENTRIES,
-  RIGHTS.EXIST_PID,
-  RIGHTS.UNSUBSCRIBE_NEWSLETTER,
-  RIGHTS.SUBSCRIBE_NEWSLETTER,
-  RIGHTS.TRANSACTION_LIST,
-  RIGHTS.SEND_COINS,
-  RIGHTS.LOGOUT,
-  RIGHTS.UPDATE_USER_INFOS,
-  RIGHTS.HAS_ELOPAGE,
-  RIGHTS.CREATE_TRANSACTION_LINK,
-  RIGHTS.DELETE_TRANSACTION_LINK,
-  RIGHTS.REDEEM_TRANSACTION_LINK,
-  RIGHTS.LIST_TRANSACTION_LINKS,
-  RIGHTS.GDT_BALANCE,
-  RIGHTS.CREATE_CONTRIBUTION,
-  RIGHTS.DELETE_CONTRIBUTION,
-  RIGHTS.LIST_CONTRIBUTIONS,
-  RIGHTS.LIST_ALL_CONTRIBUTIONS,
-  RIGHTS.UPDATE_CONTRIBUTION,
-  RIGHTS.SEARCH_ADMIN_USERS,
-  RIGHTS.LIST_CONTRIBUTION_LINKS,
-  RIGHTS.COMMUNITY_STATISTICS,
-  RIGHTS.CREATE_CONTRIBUTION_MESSAGE,
-  RIGHTS.LIST_ALL_CONTRIBUTION_MESSAGES,
-  RIGHTS.OPEN_CREATIONS,
-  RIGHTS.USER,
+  ...USER_RIGHTS,
+])
+export const ROLE_MODERATOR = new Role(ROLE_NAMES.ROLE_NAME_MODERATOR, [
+  ...INALIENABLE_RIGHTS,
+  ...USER_RIGHTS,
+  ...MODERATOR_RIGHTS,
+])
```

Comment:
Ich würde erwarten dass die ``MODERATOR_RIGHTS`` schon ``INALIENABLE_RIGHTS`` & ``USER_RIGHTS`` beinhaltet.
So müsste hier nur noch:
```suggestion
export const ROLE_MODERATOR = new Role(ROLE_NAMES.ROLE_NAME_MODERATOR, [...MODERATOR_RIGHTS])
```